soc/xtensa/intel_adsp/cavs/power.c

```diff
@@ -76,6 +79,51 @@ static inline void __sparse_cache *uncache_to_cache(void *address)
 	return (void __sparse_cache *)((uintptr_t)(address) | SRAM_ALIAS_OFFSET);
 }
 
+static ALWAYS_INLINE void _save_core_context(void)
+{
+	uint32_t core_id = arch_proc_id();
+
+	core_desc[core_id].excsave2 = XTENSA_RSR(ZSR_CPU_STR);
```

Comment:
This is correct, but calling the field "excsave2" is correct, but semantically fragile.  All it takes is for someone to come along and change the register ordering in gen_zsr.py and we'll select a different register to store in that field.  Would personally want to see this called "cpu" or "cpu_ptr" or whatever, but that's just a nitpick.